src/term.rs

```diff
@@ -93,6 +96,11 @@ pub enum Term {
         HashMap<Ident, RichTerm>,
         Vec<(RichTerm, RichTerm)>, /* field whose name is defined by interpolation */
         RecordAttrs,
+        // Cached free variables calculation
+        // An empty `Ident` is used for all dynamic fields.
```

Comment:
So don't track dyn fields free variables set?
I considered trying to name them by position, but I'm not sure there ordering is stable.

If we don't need to track them then we don't need `VecDeque`.

Reply:
> So don't track dyn fields free variables set?

To first approximation, I would say the opposite: at this point, that's not much harder to track them like any statically named field, and that's the more logical thing to do. Indexing by position is fine, as not only the order should be stable across all the transformations the AST go through, but here we are moreover talking about stability between two call to `traverse` on consecutive nodes. You could have a separate `Vec` for that in the field of `RecRecord`, say that would now be `(HashMap<_, _>, Vec<_>)`. Putting everything in the hashmap might be trickier than it first appears to, because you can define arbitrary record name `{"__%1" = "mystring"}`, so you have to handle conflicts with your naming schema.

Reply:
Sorry, that's wrong, we indeed need the order of dynamic fields to be stable between the free vars transformation and the evaluation. This is currently the case, I believe, and I can't see right now a good reason this wouldn't. But it's true that it's a new invariant to document and to maintain.